tests/integration/service_process.py

```diff
@@ -0,0 +1,316 @@
+        """
+        Start the service subprocess.
+
+        Parameters
+        ----------
+        startup_delay:
+            Time to wait after starting the service for it to initialize (seconds)
+        """
+        # Build command line arguments
+        args = [sys.executable, '-m', self.service_module]
+
+        # Add log level to command line
+        args.extend(['--log-level', self.log_level])
+
+        for key, value in self.kwargs.items():
+            # Convert Python naming to CLI flags (underscore to hyphen)
+            flag_name = key.replace('_', '-')
+
+            if isinstance(value, bool):
+                # Boolean flags: only add if True
+                if value:
+                    args.append(f'--{flag_name}')
+            else:
+                # Regular arguments
+                args.extend([f'--{flag_name}', str(value)])
+
+        logger.info("Starting service: %s with args: %s", self.service_module, args)
+
+        # Start subprocess with pipes for stdout/stderr
+        self.process = subprocess.Popen(  # noqa: S603
+            args,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            text=True,
+            bufsize=1,  # Line buffered
+        )
+
+        # Start threads to stream output
+        self._stop_event.clear()
+        self._stdout_thread = threading.Thread(
+            target=self._stream_output,
+            args=(self.process.stdout, self._stdout_lines, 'stdout'),
+            daemon=True,
+        )
+        self._stderr_thread = threading.Thread(
+            target=self._stream_output,
+            args=(self.process.stderr, self._stderr_lines, 'stderr'),
+            daemon=True,
+        )
+        self._stdout_thread.start()
+        self._stderr_thread.start()
+
+        # Wait for service to report it has started (or crash)
+        start_time = time.time()
+        service_started = False
+        while time.time() - start_time < startup_delay:
+            # Check if process crashed
+            if self.process.poll() is not None:
+                # Wait for threads to finish reading any error output
+                self._stop_event.set()
+                if self._stdout_thread:
+                    self._stdout_thread.join(timeout=1.0)
+                if self._stderr_thread:
+                    self._stderr_thread.join(timeout=1.0)
+
+                stdout = ''.join(self._stdout_lines)
+                stderr = ''.join(self._stderr_lines)
+                raise RuntimeError(
+                    f"Service {self.service_module} failed to start.\n"
+                    f"Exit code: {self.process.returncode}\n"
+                    f"STDOUT: {stdout}\n"
+                    f"STDERR: {stderr}"
+                )
+
+            # Check if we've seen "Service started" in output
+            combined_output = ''.join(self._stdout_lines + self._stderr_lines)
+            if 'Service started' in combined_output:
```

Comment:
This seems brittle. Is it part of the contract for services to output this message? Do services have a heartbeat or some kind of alive 'endpoint'?